### ISSUE TITLE
Editor does not show diagnostic errors in tooltips

### ISSUE THREAD

**Issue by marjan-georgiev** [atularen/ngx-monaco-editor#69]
The editor does not seem to be displaying the diagnostic errors in the tooltip when you have the validate diagnosticsOptions set to true.

You can reproduce this by setting the language on this line to 'json':
https://github.com/atularen/ngx-monaco-editor/blob/master/src/app/app.component.ts#L77

The JSON diagnostic options are already set to validate in app.module.ts.

This is what you see when you hover over an error in the editor:
![image](https://user-images.githubusercontent.com/8687/43050086-7402108a-8e03-11e8-8b62-8183320ac1a0.png)

It says `undefined`. Instead, it should display the actual error. I am seeing this for all languages, so the problem is not with the language itself - it is returning the correct error.

This is working fine in the `monaco-editor` demo:

![image](https://user-images.githubusercontent.com/8687/43050116-378ebe7c-8e04-11e8-8293-ccd73b9b216c.png)

Any idea where the issue might be and how to debug it?

**Comment by king2bob** [comment]
I took a stab at trying to investigate this and haven't got very far.
I'm running Chrome 67 on Windows with monaco 0.13.1.

I managed to isolate part of the problem. It seems that [angular's zone.js](https://github.com/angular/zone.js) is interfering with monaco's promises.

Here is a [a fiddle](https://jsfiddle.net/gqb0om41/) that also has the same issue

![image](https://user-images.githubusercontent.com/5145738/43572708-0da8352a-9638-11e8-9c48-3944ba92232d.png)

but if the monaco loader is included before [zone.js](https://github.com/angular/zone.js) then it seems to work as expected
[here is another fiddle](https://jsfiddle.net/cqso90d4/) but with the resources swaped

![image](https://user-images.githubusercontent.com/5145738/43572845-5a365552-9638-11e8-9ea9-e40e509d46a2.png)

![image](https://user-images.githubusercontent.com/5145738/43572900-7841c5cc-9638-11e8-944e-469ac3497377.png)

Digging a little bit deeper I was able to load [zone.js](https://github.com/angular/zone.js) first only if I disabled Zone aware promises (as detailed in the [Module docs](https://github.com/angular/zone.js/blob/master/MODULE.md)) 

`__Zone_disable_ZoneAwarePromise = true`

[This fiddle](https://jsfiddle.net/yL0zqgb3/) has this setup but for most angular 2+ setups you'll probably need Zone aware promises enables as it is important for change detection. 

Here is an excerpt from [the Module docs](https://github.com/angular/zone.js/blob/master/MODULE.md)

>- Angular(2+)
>
> Angular uses zone.js to manage async operations and decide when to perform change detection. Thus, in Angular, the following APIs should be patched, otherwise Angular may not work as expected.
>
> 1. ZoneAwarePromise
> 2. timer
> 3. on_property
> 4. EventTarget
> 5. XHR

I'm yet to find the right order of script loads to make it work inside an angular app without breaking one or the other.

**Comment by marjan-georgiev** [comment]
@king2bob that's a great find! Did you try running it outside the angular zone, and only running the callbacks inside of it?

**Comment by king2bob** [comment]
@marjan-georgiev I tried that initially but it did not seem to help.
`runOutsideAngular` seems to run in a zone just note the one with the angular change-detection

https://github.com/angular/angular/blob/c8a1a14b87e5907458e8e87021e47f9796cb3257/packages/core/src/zone/ng_zone.ts#L202-L218

From the zone.js side it doesn't seem like there is a way to run some in a zone and some not because it monkey-patches the native/browser functions. There is no choice after it has been loaded in and the `__Zone+*` flags seem to be all you can do to change things.

**Comment by king2bob** [comment]
here is [a stackblitz](https://stackblitz.com/edit/angular-jdrneq) based on this repo
![image](https://user-images.githubusercontent.com/5145738/43642212-95cdf6ec-971e-11e8-86bc-1cb28e382c83.png)

and [another stackblitz](https://stackblitz.com/edit/angular-sntxjr) with all the step running inside a call to `runOutsideAngular`

![image](https://user-images.githubusercontent.com/5145738/43642220-9dcd0e32-971e-11e8-86f8-2ee31e22d8cb.png)

no joy I'm afraid.

**Comment by king2bob** [comment]
It turns out I should have spent more time in the monaco-editor issues page....
[This issue](https://github.com/Microsoft/monaco-editor/issues/790) is exactly the problem and the workaround 
```
Promise.all = function (values: any): Promise<any> {
    let resolve: (v: any) => void;
    let reject: (v: any) => void;
    let promise = new this((res, rej) => {
      resolve = res;
      reject = rej;
    });
    let count = 0;
    let index = 0;
    const resolvedValues: any[] = [];
    for (let value of values) {
      if (!(value && value.then)) {
        value = this.resolve(value);
      }
      value.then(
          ((index) => (value: any) => {
            resolvedValues[index] = value;
            count--;
            if (!count) {
                resolve(resolvedValues);
            }
          })(index),
          reject);
      count++;
      index++;
    }
    if (!count) resolve(resolvedValues);
    return promise;
}
```
seems to work.

See here 
https://stackblitz.com/edit/angular-5qhtyz?file=src%2Fpolyfills.ts

Not sure if there is a smarter way to add this into the project. It seems to boil down to overwriting the WinJS version of `Promise.all` after zone.js has load.

Anyway I hope this helps anyone who stumbles onto this.

**Comment by marjan-georgiev** [comment]
I can confirm that fixes it. Thanks for finding that, I thought it was specific to this package so I didn't think to look there.

**Comment by Louis-7** [comment]
Hope this can be fix in future version of monaco-editor.

**Comment by marjan-georgiev** [comment]
This seems to be fixed in zone.js with this: https://github.com/angular/zone.js/pull/1150

We should be able to close the issue.

**Comment by king2bob** [comment]
Yep that seems to have solved it. Another [stackblitz](https://stackblitz.com/edit/angular-fmzu3n) for conformation using Angular 7.2.5.
![image](https://user-images.githubusercontent.com/5145738/55682079-1fe15080-5926-11e9-9300-0fed2654eeb2.png)

Thanks to @marjan-georgiev for checking back in.
